xarray/tests/test_parallelcompat.py

```diff
@@ -0,0 +1,163 @@
+from xarray.tests import requires_dask
+
+
+class DummyChunkedArray(np.ndarray):
+    """
+    Mock-up of a chunked array class.
+
+    Adds a (non-functional) .chunks attribute by following this example in the numpy docs
+    https://numpy.org/doc/stable/user/basics.subclassing.html#simple-example-adding-an-extra-attribute-to-ndarray
+    """
+
+    chunks: Optional[T_Chunks]
+
+    def __new__(
+        cls,
+        shape,
+        dtype=float,
+        buffer=None,
+        offset=0,
+        strides=None,
+        order=None,
+        chunks=None,
+    ):
+        obj = super().__new__(cls, shape, dtype, buffer, offset, strides, order)
+        obj.chunks = chunks
+        return obj
+
+    def __array_finalize__(self, obj):
+        if obj is None:
+            return
+        self.chunks = getattr(obj, "chunks", None)
+
+    def rechunk(self, chunks, **kwargs):
+        copied = self.copy()
+        copied.chunks = chunks
+        return copied
+
+
+class DummyChunkManager(ChunkManagerEntrypoint):
+    """Mock-up of ChunkManager class for DummyChunkedArray"""
+
+    def __init__(self):
+        self.array_cls = DummyChunkedArray
+
+    def is_chunked_array(self, data: Any) -> bool:
+        return isinstance(data, DummyChunkedArray)
+
+    def chunks(self, data: DummyChunkedArray) -> T_Chunks:
+        return data.chunks
+
+    def from_array(
+        self, data: np.ndarray, chunks: T_Chunks, **kwargs
+    ) -> DummyChunkedArray:
+        from dask import array as da
+
+        return da.from_array(data, chunks, **kwargs)
+
+    def rechunk(self, data: DummyChunkedArray, chunks, **kwargs) -> DummyChunkedArray:
+        return data.rechunk(chunks, **kwargs)
+
+    def compute(self, *data: DummyChunkedArray, **kwargs) -> np.ndarray:
+        from dask.array import compute
+
+        return compute(*data, **kwargs)
+
+    def apply_gufunc(
+        self,
+        func,
+        signature,
+        *args,
+        axes=None,
+        axis=None,
+        keepdims=False,
+        output_dtypes=None,
+        output_sizes=None,
+        vectorize=None,
+        allow_rechunk=False,
+        meta=None,
+        **kwargs,
+    ):
+        from dask.array.gufunc import apply_gufunc
+
+        return apply_gufunc(
+            func,
+            signature,
+            *args,
+            axes=axes,
+            axis=axis,
+            keepdims=keepdims,
+            output_dtypes=output_dtypes,
+            output_sizes=output_sizes,
+            vectorize=vectorize,
+            allow_rechunk=allow_rechunk,
+            meta=meta,
+            **kwargs,
+        )
+
+
+@pytest.fixture
+def register_dummy_chunkmanager():
+    """Mocks the registering of an additional ChunkManagerEntrypoint."""
+    EXAMPLE_CHUNKMANAGERS["dummy"] = DummyChunkManager
+    yield
+    del EXAMPLE_CHUNKMANAGERS["dummy"]
+
+
+class TestGetChunkManager:
+    def test_get_chunkmanger(self, register_dummy_chunkmanager):
+        chunkmanager = guess_chunkmanager("dummy")
```

Comment:
try using `monkeypatch.setattr`  (`monkeypatch` is a built-in fixture) for this. With that, you put a empty dict (or a dict containing just the dummy chunk manager) in place of the global dict, and it also makes sure that this change is limited in scope (so you don't accidentally influence one test from another)

Reply:
alternatively use `with unittest.mock.patch.dict(..., newdict): yield`
maybe this works?
But I don't see a reason why the current approach should _not_ work

Reply:
actually, just `monkeypatch` out `list_chunkmanagers`, that should be much easier

Reply:
> But I don't see a reason why the current approach should not work

I'm guessing that we ran into race conditions since it modified global state (we're executing multiple tests in parallel using `pytest-xdist`).